### PR TITLE
Create GPS Subsystem.md

### PR DESCRIPTION
**Pull request by JoshuaEgwuatu**

### CODE REVIEW

**Review by JesseTNRoberts — COMMENTED** [review]
These constraints aren't precise enough to be design criteria

File: Environmental Sensing/GPS Subsystem.md [hunk 1]
```diff
@@ -0,0 +1,219 @@
+# Function of the subsystem:
```
Comment (JesseTNRoberts) [inline Environmental Sensing/GPS Subsystem.md:1]: Why are we discussing component selection here?

File: Environmental Sensing/GPS Subsystem.md [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+The robot will use a GPS subsystem to track its location in real-time, which will help improve environmental data collection. By integrating GPS, the robot can continuously determine its precise coordinates, enhancing the accuracy and efficiency of data collection. It can navigate to specific locations based on GPS coordinates, allowing for targeted data collection from different points of interest. This approach enables more precise data gathering.
+
+The team chose the M10Q-5883 GNSS GPS & Compass Module, for the function of real-time location tracking to improve environmental data collection in a robot.
+
+The GPS subsystem works with the top-level controller and robotic operating system (ROS). The controller manages different parts of the robot and can use GPS data along with other sensors to make decisions and control its actions. ROS helps the GPS subsystem connect with other systems like mapping and path planning, making everything work together smoothly.
+
+## Function:
```
Comment (JesseTNRoberts) [inline Environmental Sensing/GPS Subsystem.md]: This is way too long for a description of what this system is to do. Be brief here.

File: Environmental Sensing/GPS Subsystem.md [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+
+Given the length of 16.5 ft and width of 6 ft, the total area of the crawlspace is 99 square feet. By using floor tiles as a reference, with each tile measuring 9 in by 9 in, we can determine that the crawlspace occupies an area of 11 by 11 tiles. The walls of the model crawlspace, constructed with various materials, have a minimum height of 17 in and a maximum height of 36 in. Considering an average height of 26.5 in (2.21 ft), we can calculate the area of the walls. Assuming the walls are straight and uniformly tall, the area of the walls is estimated to be 99.45 square feet.
+
+> "rectangular-shaped model crawlspace was created, with a length of 16.5 ft, a width of 6 ft, and a resulting area of 99 square feet. Floor tiles were used as a reference for quantifying the crawlspace area, with each floor tile having an area of 9 in by 9 in. The walls of the model crawlspace were made with various materials available in the laboratory and had a minimum height of 17 in and a maximum height of 36 in." from the Experimental Data
+
+
+
+| No.|  Specifications and Constraints                                                                                             |
```
Comment (JesseTNRoberts) [inline Environmental Sensing/GPS Subsystem.md]: These aren't precise. There is not way to design to these constraints. Further, there are constraints that haven't been identified here that are appropriate.

**Review by JesseTNRoberts — COMMENTED** [review]
There's no way to design or analyze without first having a firm idea of what the constraints are.

**Review by JesseTNRoberts — COMMENTED** [review]
Did not review. Previous comments have not been addressed.

File: Environmental Sensing/GPS Subsystem.md [hunk 4]
```diff
@@ -0,0 +1,247 @@
+# Function of the subsystem:
+
+By incorporating a GPS and compass device into the Autonomous Crawlspace Inspection Robot, it would have the capability to accurately determine its position within the crawlspace. This integration of GPS would enable the robot to generate a map of the crawlspace, displaying its current location and the areas it has explored. By combining GPS and compass data, the robot can plan efficient inspection routes and avoid revisiting previously inspected areas. This optimization ensures a comprehensive inspection while minimizing unnecessary movements. Additionally, by correlating the GPS and compass data with sensor readings or captured images, the collected data can be precisely tagged with location information.
```
Comment (JesseTNRoberts) [inline Environmental Sensing/GPS Subsystem.md]: I can't review this without understanding what this subsystem is supposed to do.

Tell me only what functionality you expect this to have. Don't write a paragraph. Write a single sentence that says what this system's purpose is. What do you expect it to do?

**Review by JesseTNRoberts — COMMENTED** [review]
More words aren't necessarily better. Say exactly enough to get your point across and no more.

Your analysis should address exactly the system constraints. The best thing to do is to number the constraints and address them in the same order.

File: Environmental Sensing/GPS Subsystem.md [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+
+## Constraint 2: 
+The acquired GPS readings shall meet the following accuracy criteria:
+   - Horizontal accuracy: within 3 meters.
+   - Vertical accuracy: within 5 meters.
+   - Time accuracy: within 30 nanoseconds.
+   - Speed accuracy: 0.006 m/sec or less 
+
+
+## Constraint 3:
```
Comment (JesseTNRoberts) [inline Environmental Sensing/GPS Subsystem.md]: This isn't a constraint

File: Environmental Sensing/GPS Subsystem.md [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+The percentage of signal blockage caused by the walls is (99.45 square feet / 99 square feet) x 100 ≈ 100.45%.
+Given the length of 16.5 ft and width of 6 ft, the total area of the crawlspace is 99 square feet. By using floor tiles as a reference, with each tile measuring 9 in by 9 in, we can determine that the crawlspace occupies an area of 11 by 11 tiles. The walls of the model crawlspace, constructed with various materials, have a minimum height of 17 in and a maximum height of 36 in. Considering an average height of 26.5 in (2.21 ft), we can calculate the area of the walls. Assuming the walls are straight and uniformly tall, the area of the walls is estimated to be 99.45 square feet.
+
+> "rectangular-shaped model crawlspace was created, with a length of 16.5 ft, a width of 6 ft, and a resulting area of 99 square feet. Floor tiles were used as a reference for quantifying the crawlspace area, with each floor tile having an area of 9 in by 9 in. The walls of the model crawlspace were made with various materials available in the laboratory and had a minimum height of 17 in and a maximum height of 36 in." from the Experimental Data
+
+
+
+# Analysis
```
Comment (JesseTNRoberts) [inline Environmental Sensing/GPS Subsystem.md]: Analysis comes after the design. Reorder.

**Review by JesseTNRoberts — COMMENTED** [review]
There are many things in this signoff that need to be trimmed. Make sure it is the proper order. The schematic is not correct and needs more work. Specifically, this needs to be integrated into the larger circuit for the bot.

**Review by JesseTNRoberts — COMMENTED** [review]
Probably need to reconsider the use of the GPS device sense we won't get signal. From discussion after class.

**Review by JesseTNRoberts — COMMENTED** [review]
I think that needing 3 repeaters make this to unwieldy to expect someone to use. Again, let's look at a magnetic compass solution.

File: Environmental Sensing/GPS Subsystem.md [hunk 7]
```diff
@@ -0,0 +1,260 @@
+# Function of the subsystem:
+
+The GPS system is expected to provide real-time location tracking of the autonomous crawlspace inspection robot to enhance the accuracy of environmental data collection.
```
Comment (JesseTNRoberts) [inline Environmental Sensing/GPS Subsystem.md]: This isn't the desired function we discussed.

File: Environmental Sensing/GPS Subsystem.md [hunk 8]
```diff
@@ -0,0 +1,260 @@
...
+
+
+# Constraints:
+
+## Constraint 1: 
+The robot shall be capable of acquiring a reliable GPS signal within a time limit of less than 0.06 seconds, considering limited sky visibility and potential signal blockage.
```
Comment (JesseTNRoberts) [inline Environmental Sensing/GPS Subsystem.md]: Why?

**Review by JesseTNRoberts — COMMENTED** [review]
We discussed in class that using a GPS was not a good option. We have been having the same conversation for 3 weeks now. You must propose a design that may function. You correctly established very early on that GPS would not work. So, I strongly suggest that you update the intended function (the objective is not real time location) and propose a solution that does not rely on GPS.

File: Environmental Sensing/GPS Subsystem.md [hunk 9]
```diff
@@ -0,0 +1,142 @@
+# Function of the subsystem:
+Using the compass enables the robot to determine its orientation and optimize path planning.
```
Comment (JesseTNRoberts) [inline Environmental Sensing/GPS Subsystem.md]: How does it permit path planning optimization? 

Before beginning any design, be absolutely certain that you understand the purpose. The compass has no use for path planning in this context. 

The purpose we have discussed for this system is to make it possible to overlay the map generated by the existing robot software onto a satellite map of the area. You have established that GPS can't get signal and won't be viable here. Good. So, how can it be done?

Without knowing what the system is precisely intended to do, it is impossible to develop constraints for the system. Without constraints, it is impossible to design the system.

Comment (fnalheji42) [reply Environmental Sensing/GPS Subsystem.md]: I have rewritten the function, but  I have not yet posted the updated function on GitHub

The compass subsystem continuously reads and processes data to calculate the robot's heading, enabling navigation, mapping, and overlaying the robot's map onto a satellite map in the autonomous crawlspace inspection robot.

Additionally, the constraints for this subsystem include considerations related to magnetic interference, calibration, and precision limitations.

I hope this meets the expectations of this subsystem, please let me know if this is ok

**Review by JesseTNRoberts — APPROVED** [review]

**Comment by JesseTNRoberts** [comment]
The only change is that I want to use a cheaper microcontroller. There's no need for an atmega.

**Pull request merged**